src/Functions/FunctionsConversion.h

```diff
@@ -1705,12 +1705,21 @@ struct ConvertImpl
                 else if constexpr (std::is_same_v<FromDataType, DataTypeIPv6> && std::is_same_v<ToDataType, DataTypeUInt128>)
                 {
                     static_assert(
-                        std::is_same_v<DataTypeUInt128::FieldType, DataTypeUUID::FieldType::UnderlyingType>,
+                        std::is_same_v<DataTypeUInt128::FieldType, DataTypeIPv6::FieldType::UnderlyingType>,
                         "UInt128 and IPv6 types must be same");
 
                     vec_to[i].items[1] = std::byteswap(vec_from[i].toUnderType().items[0]);
                     vec_to[i].items[0] = std::byteswap(vec_from[i].toUnderType().items[1]);
                 }
+                else if constexpr (std::is_same_v<FromDataType, DataTypeUInt128> && std::is_same_v<ToDataType, DataTypeIPv6>)
+                {
+                    static_assert(
+                        std::is_same_v<DataTypeUInt128::FieldType, DataTypeIPv6::FieldType::UnderlyingType>,
+                        "IPv6 and UInt128 types must be same");
+
+                    vec_to[i].toUnderType().items[1] = std::byteswap(vec_from[i].items[0]);
```

Comment:
I'm not sure about this.

Reply:
I wasn't sure about this either, initially. I actually copied this from here: https://github.com/ClickHouse/ClickHouse/pull/76928/files#diff-2859e9a878020ac61a5586d7413871dfc37b790071c191bc1fe3f091ee56370eR1711-R1712

After my testing, I realized that IPv6 in clickhouse is stored in 16 bytes as UInt128 big-endian, so we need to reverse the byte order from UInt128's default (at least little-endian for linux).
https://clickhouse.com/docs/sql-reference/data-types/ipv6

I added a few more tests to catch the byte order issues, as well